src/development/platform-integration/ios/apple-frameworks.md

```diff
@@ -0,0 +1,200 @@
+
+<!-- Add once SwiftUI PR is merged -->
+<!-- For an overview of how the SwiftUI framework compares to Flutter, 
+see [Learning Flutter as a SwiftUI developer][].  -->
+
+### Flutter plugin overview
+When developing an app with Flutter, you can interact 
+with system libraries using plugins. 
+Plugins are Dart libraries that contain platform-specific code. 
+In the case of an iOS app, 
+the platform code can be written in Objective-C or Swift. 
+
+In your Dart code, you use the plugin's Dart API and 
+behind the scenes the native plugin code will directly call 
+the system library being used. This means that you can write 
+the code to call the Dart API once, 
+and it will work for all the platforms 
+that the plugin has been written for. 
+
+To learn more about plugins see [Using packages][]. 
+This document links out to some commonly used plugins, 
+and their examples, but there are thousands more available on [pub.dev][]. 
+And, if you can't find what you're looking for, you can create your own. 
+
+### Adding a plugin to your project
+When using Apple frameworks within your native project, 
+you usually just need to import it into your Swift or Objective-C file. 
+
+In Flutter, you must first run `flutter pub add package_name` 
+in the terminal, from the root of your project. This adds 
+the dependency to your [`pubspec.yaml`][] file. You can then import 
+the package in your Dart file. 
+
+Occasionally, you may also need to change app settings or 
+initialization logic- which will be explained in the package's 
+"Readme" page on [pub.dev][].
+
+### Accessing the photo library
+In Swift and Objective-C, you use `PhotoKit` and the `PhotosUI` framework
+to access user photos.
+
+In Flutter, you can use the [`image_picker`][] plugin, 
+which supplies an `ImpagePicker` class that is used to 
+display the user's photo library.
+
+### Accessing the camera
+In SwiftUI and UIKit, to allow your app's users to take a photo, 
+you use the `UIImagePickerController` class, 
+with the `.camera` `sourceType`.
+
+In Flutter, to access the camera, 
+you'll use the same [`image_picker`][] plugin that was 
+referenced above. The code will look similar, 
+only now the `source` for `pickImage` is the camera.
+
+### In-app purchases
+In Swift and Objective-c, you use `StoreKit` to 
+offer in-app purchases.
+
+In Flutter, you can support in-app purchases through the 
+Google Play store on Android and the 
+Apple App Store on iOS by using the 
+[`in_app_purchase`][] plugin.
+
+### Payment processing
+In Swift and Objective-C, you use `PassKit` to process 
+Apple Pay payments in your app. 
+
+In Flutter, you can use the [`pay`][] plugin to 
+add Google Pay payments on Android
+ and Apple pay payments on iOS.
+
+### Push notifications
+In Swift and Objective-C, you use the `UserNotifications` 
+framework for pushing user-facing notifications.
+
+In Flutter, you can use the [`firebase_messaging`][] plugin, 
+which uses Firebase Cloud Messaging. 
```

Comment:
Is there a plugin that supports APNS?

Reply:
It seems like there is but it's from an unverified publisher so not sure if it's worth mentioning: https://pub.dev/packages/flutter_apns

I haven't personally used FCM before but it seems like you may be able to use APNs through their integration: https://firebase.flutter.dev/docs/messaging/apple-integration/

Reply:
ah gotcha. maybe worth mentioning, since APNS may be what devs are looking for, if they have all their severs setup already.

Reply:
Good point thanks!